internal/pkg/otel/manager/execution_embedded.go

```diff
@@ -0,0 +1,80 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package manager
+
+import (
+	"context"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/status"
+	"go.opentelemetry.io/collector/confmap"
+	"go.opentelemetry.io/collector/otelcol"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+
+	"github.com/elastic/elastic-agent/internal/pkg/otel"
+	"github.com/elastic/elastic-agent/internal/pkg/otel/agentprovider"
+	"github.com/elastic/elastic-agent/internal/pkg/release"
+	"github.com/elastic/elastic-agent/pkg/core/logger"
+)
+
+func newExecutionEmbedded() *embeddedExecution {
+	return &embeddedExecution{}
+}
+
+type embeddedExecution struct {
+}
+
+// startCollector starts the collector in a new goroutine.
+func (r *embeddedExecution) startCollector(ctx context.Context, logger *logger.Logger, cfg *confmap.Conf, errCh chan error, statusCh chan *status.AggregateStatus) (collectorHandle, error) {
+	collectorCtx, collectorCancel := context.WithCancel(context.Background())
```

Comment:
The sub-process is wired into the parent context but this one isn't, why not? It makes the `Stop` call redundant if the context is already cancelled but that is true of the sub-process case too.